#!/usr/bin/env python
PACKAGE = "sentry_chassis_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 轮子速度PID参数
gen.add("wheel_kp", double_t, 0, "轮子Kp", 2.0, 0.0, 500.0)
gen.add("wheel_ki", double_t, 0, "轮子Ki", 0.0, 0.0, 10.0)
gen.add("wheel_kd", double_t, 0, "轮子Kd", 0.0, 0.0, 10.0)

# 转向角度PID参数
gen.add("pivot_kp", double_t, 0, "转向Kp", 100.0, 0.0, 500.0)
gen.add("pivot_ki", double_t, 0, "转向Ki", 0.0, 0.0, 10.0)
gen.add("pivot_kd", double_t, 0, "转向Kd", 1.0, 0.0, 10.0)

# 新增：速度模式参数
gen.add("use_global_vel", bool_t, 0, "速度模式：true=全局坐标系，false=底盘坐标系", False)
# 新增：自锁角度参数（弧度）
gen.add("lock_angle", double_t, 0, "自锁角度（弧度）", 0.785, 0.0, 1.57)
gen.add("enable_lock", bool_t, 0, "启用自锁功能", True)


exit(gen.generate(PACKAGE, "sentry_chassis_controller", "SentryChassisController"))
